### ISSUE TITLE
Starting PWAs causes spurious tabs to open in main browser

### ISSUE THREAD

**Issue by shawnz** [filips123/PWAsForFirefox#349]
### Description

Hi there, after installing FirefoxPWA, creating some PWAs and then logging into Firefox Sync in the PWA profile, I now get spurious tabs opening in my main browser whenever I launch a PWA. For example, I just launched my IRCCloud PWA and a few minutes later a tab suddenly opened in my browser pointing to the Canada Post website (which I visited recently).

### Steps to Reproduce

1. Install FirefoxPWA
2. Create some PWAs
3. Launch the PWAs and sign into Firefox Sync in the PWA profile
4. Wait

### Environment

* Operating system: Windows 11 22H2
* System architecture: x86
* Installation method: MSI
* PWAsForFirefox extension version: 2.6.0
* PWAsForFirefox native version: 2.6.0
* Firefox version: Nightly 115.0a1

**Comment by filips123** [comment]
Do you have "Open out-of-scope URLs in a default browser" enabled?

**Comment by shawnz** [comment]
Yes, I have that option enabled as well.

**Comment by filips123** [comment]
Can you disable it and try again? Also, does it always happen and with all PWAs?

This issue happens because those websites are out-of-scope so they are opened in your default browser. However, I don't know why they are being opened automatically in the first place.

**Comment by emptyother** [comment]
Try this: In your main browser, go to about:newtab and disable both feeds. Then let it sync. Open a PWA, and let it sync. Now close all firefox instances, and then open a PWA and see if it still happens.

For me it stops if those newtab feeds are disabled, it seems.

**Comment by filips123** [comment]
You can also go to the new tab page if you press F6 and enter `about:newtab`. This way you won't have to use sync to change this setting.

I still don't know what would cause tabs to be opened though. Do those tabs also open (but in the PWA) if you disable "Open out-of-scope URLs in a default browser"? And if so, what do they contain?

Also, another thing... Which settings exactly did you disable? I'm probably not in the region where all those feeds are enabled, so I'm not sure what exactly should be disabled. If you can give me a list of all settings that should be disabled, I can try to disable them by default.

**Comment by shawnz** [comment]
Sorry for the delay, I'm testing now with the "out-of-scope URLs" option disabled and haven't had any popups yet. However I would like to leave that option enabled if possible.

The tabs which open are always websites that I visited recently.

Regarding the new tab page feeds, I'm not sure what you mean. I don't see any option like that on the new tab page.

**Comment by filips123** [comment]
Maybe it has something to do with restoring previous sessions. Do you have "Open previous windows and tabs" enabled in settings of the PWA browser? If so, try to disable it.

**Comment by shawnz** [comment]
"Open previous windows and tabs" is already disabled for me

**Comment by filips123** [comment]
Unfortunately, then I don't know what causes this. Maybe check those new tab feeds and related settings (maybe recent activity, snippets, shortcuts, etc.). If you want, you can also try to search through `about:config` and settings for any settings related to the session and opening previous tabs, but that's about everything that I can currently think of that could be related to this problem.

**Comment by emptyother** [comment]
> Which settings exactly did you disable?

If these two is set to `false` the sites stop (but only after I've restarted the FirefoxPWA, even if I used F6 to open either `about:config` or `about:newtab`). 

- `browser.newtabpage.activity-stream.feeds.section.highlights`
- `browser.newtabpage.activity-stream.feeds.topsites`

> Do those tabs also open (but in the PWA) if you disable "Open out-of-scope URLs in a default browser"?

No.

> what do they contain?

They mostly contain links from my pinned frontpage and recent history. But one or another time it opens a picture from a recent website instead.

Also they opens every 10th second or so. Also it eventually stops after about 30 tabs.

I would assume you can fix it by setting those settings above to false in FirefoxPWA, while still having them stay user value on regular firefox, if thats possible. Or change FirefoxPWAs newtab page to a blank site? Its not like FirefoxPWA need the newtab page.

**Comment by filips123** [comment]
> I would assume you can fix it by setting those settings above to false in FirefoxPWA, while still having them stay user value on regular firefox, if thats possible.

Yes, I can set those settings to false. I think they won't sync with with regular Firefox, because they are configured as default values. I'll probably both disabe those preferences and change newtab and home pages, just in case...

> But one or another time it opens a picture from a recent website instead.

Interesting, maybe this is also a similar issue as #253. I still don't know why both of this issues happen, but if changing those settings fixes them, great.

**Comment by filips123** [comment]
I've disabled some activity stream feeds and also disabled home and new tab pages in [2.7.0](https://github.com/filips123/PWAsForFirefox/releases/tag/v2.7.0). Please check if this fixes the issue.

**Comment by emptyother** [comment]
No, unfortunately it keeps happening with the `firefoxpwa-2.7.0-x86_64.msi` installed. Darn weird. Thanks for trying though.

**Comment by filips123** [comment]
I've just thought about something... Can you check if those preferences (that I've disabled in [this commit](https://github.com/filips123/PWAsForFirefox/commit/0aed124df303c3899e1c69de3ade0fbf0a61c3a4#diff-1c631791e119c50eb41bb25b00d9d5fb34b0531aba49e27efe0592bc6392d8e9R1830-R1835)) are actually disabled for you? They are only changed to false as default values, so if you used sync (or manually changed them in the past), they might still be set to true as a user value.

**Comment by filips123** [comment]
Then it probably still happens for some users because only default values have been set to false. I'll try to also force them to false as user values and prevent them from being synced. And while I'm doing this, it might also be good to check if there are any other (existing) preferences that I modified that shouldn't sync.

**Comment by filips123** [comment]
I now released 2.7.1 with this change. I prevented syncing some preferences that might cause problems and also reset some of them back to default values.

Due to how this reset is set up, you might have to launch the web app twice before they get properly reset. Additionally, you can also try to temporarily enable "Always patch runtime and profile" before launching the web app to be more sure that the changes got copied to the profile.

**Comment by emptyother** [comment]
That 2.7.1 fixed it! 😊

**Comment by filips123** [comment]
Nice. I'll close this issue, and if this problems still happens in the future, please let me know so I can reopen it.
